client/state/sites/selectors.js

```diff
@@ -550,3 +554,304 @@ export function getSitePostsPage( state, siteId ) {
+		const isModuleActive = isJetpackModuleActive( state, siteId, 'manage' );
+		return isModuleActive === null ? true : isModuleActive;
+	}
+	// for version lower than 3.4, we cannot not determine canManage, we'll assume they can
+	return true;
+}
+
+/**
+ * Determines if a Jetpack site can update its files.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site can update its file
+ */
+export function canJetpackSiteUpdateFiles( state, siteId ) {
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! siteHasMinimumJetpackVersion( state, siteId ) ) {
+		return false;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( ! isJetpackSiteMainNetworkSite( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if ( ! fileModDisabled ) {
+		return true;
+	}
+
+	return (
+		! includes( fileModDisabled, 'disallow_file_mods' ) &&
+		! includes( fileModDisabled, 'has_no_file_system_write_access' )
+	);
+}
+
+/**
+ * Determines if a Jetpack site can auto update its files.
+ * This function checks if the given Jetpack site can update its files and if the automatic updater is enabled
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site can auto update
+ */
+export function canJetpackSiteAutoUpdateFiles( state, siteId ) {
+	if ( ! canJetpackSiteUpdateFiles( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if ( fileModDisabled && includes( fileModDisabled, 'automatic_updater_disabled' ) ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Determines if a Jetpack site can auto update WordPress core.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site can auto update WordPress
+ */
+export function canJetpackSiteAutoUpdateCore( state, siteId ) {
+	return canJetpackSiteAutoUpdateFiles( state, siteId );
+}
+
+/**
+ * Determines if the Jetpack plugin of a Jetpack Site has menus
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site has Jetpack menus management
+ */
+export function hasJetpackSiteJetpackMenus( state, siteId ) {
+	const siteJetpackVersion = getSiteOption( state, siteId, 'jetpack_version' );
+	return versionCompare( siteJetpackVersion, '3.5-alpha' ) >= 0;
+}
+
+/**
+ * Determines if the Jetpack plugin of a Jetpack Site has themes
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site has Jetpack themes management
+ */
+export function hasJetpackSiteJetpackThemes( state, siteId ) {
+	const siteJetpackVersion = getSiteOption( state, siteId, 'jetpack_version' );
```

Comment:
If we decide to go with null returns, this function should probably return `null`.